junit5-engine/src/main/java/org/junit/gen5/engine/junit5/descriptor/LifecycleMethodUtils.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2015-2016 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v1.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+
+package org.junit.gen5.engine.junit5.descriptor;
+
+import static org.junit.gen5.commons.util.AnnotationUtils.findAnnotatedMethods;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Method;
+import java.util.List;
+
+import org.junit.gen5.api.AfterAll;
+import org.junit.gen5.api.AfterEach;
+import org.junit.gen5.api.BeforeAll;
+import org.junit.gen5.api.BeforeEach;
+import org.junit.gen5.commons.JUnitException;
+import org.junit.gen5.commons.util.ReflectionUtils;
+import org.junit.gen5.commons.util.ReflectionUtils.MethodSortOrder;
+
+/**
+ * Collection of utilities for working with test lifecycle methods.
+ *
+ * @since 5.0
+ */
+final class LifecycleMethodUtils {
```

Comment:
Nice. I like the idea of having a central place of how to lookup and find the different pieces. We also do have some more of these things in extra classes. I am wondering if we could consolidate those all into a "function" library. This will not be part of this PR, just an idea I wanted to mention.